hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/HoodieSyncConfig.java

```diff
@@ -163,6 +163,11 @@ public class HoodieSyncConfig extends HoodieConfig {
       .defaultValue("")
       .withDocumentation("The spark version used when syncing with a metastore.");
 
+  public static final ConfigProperty<String> META_SYNC_PARTITION_FIXMODE = ConfigProperty
+      .key("hoodie.datasource.hive_sync.partition_fixmode")
+      .defaultValue("false")
+      .withDocumentation("Implement a full partition sync operation when partitions are lost.");
```

Comment:
`hoodie.datasource.hive_sync.partition_fixmode` -> `hoodie.datasource.hive_sync.incremental` ? And by default, this option val is true.

Reply:
if hoodie.datasource.hive_sync.partition_fixmode = true, lastCommitTimeSynced will be empty, and all hivepartitions will be obtained for comparison, and a full amount of partition metadata sync will be executed,normally it should be false

Reply:
@danny0405  Hi, take a look again？

Reply:
`fixmode` is not a good name, did quite get what it really means.

Reply:
Maybe call ‘partition_align’?  how about this?

Reply:
How about `hoodie.datasource.hive_sync.incremental` and by default is true.

Reply:
[hoodie.datasource.hive_sync.incremental] -- This name is ok. 

I think the usage scenario of this parameter is:
true :
When it is found that the partition metadata in hms is less than the actual partition of fs, an alignment needs to be done to complete the missing partitions in hms. Only when the metadata in hms needs to be completed, it needs to be opened once. When the parameter is true, lastCommitTimeSynced will be null, it is a full alignment operation at this time.
`
    // Get the last time we successfully synced partitions
    Option<String> lastCommitTimeSynced = Option.empty();
    if (tableExists & !config.getBoolean(META_SYNC_PARTITION_FIXMODE)) {
      lastCommitTimeSynced = syncClient.getLastCommitTimeSynced(tableName);
    }
`

false:
Under normal circumstances, lastCommitTimeSynced is used as the baseline to synchronize the newly generated partitions after lastCommitTimeSynced.

So the default should probably be false.

Reply:
Agree that `hoodie.datasource.hive_sync.incremental` is a better naming.  Since this can also apply to Glue Catalog sync, can we name it with `meta.sync`, i.e., `hoodie.meta.sync.incremental`?  @xushiyan what is the naming convention?  I see different prefixes are used, e.g., `hoodie.meta.sync`, `hoodie.datasource.meta_sync`, `hoodie.meta_sync`.

Reply:
If the `incremental` is the naming, default should be `true`, i.e., "lastCommitTimeSynced is used as the baseline to synchronize the newly generated partitions after lastCommitTimeSynced".

Reply:
@yihua @danny0405 
Maybe I didn't describe it clearly. The purpose of this pr is to provide a tool parameter when the metadata of the synchronization partition is found to be lost, **and the function of this parameter is not an incremental synchronization partition, but a parameter switch to control whether to use Perform a synchronous alignment operation for all partitions.**

Looking at the current code logic is to do incremental synchronization according to lastCommitTimeSynced. If set to true, according to the code, the syncAllPartitions method will be used every time to synchronize all partitions, which is unnecessary.

Reply:
@weimingdiit Understood.  I think there's a communication gap.  I revised the code based on our discussion.  Could you check if it algins with what you think?